packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -388,49 +392,69 @@ const findLane = (y1, y2, depth = 0) => {
  * Draw the lines between the commits. They were arrows initially.
  *
  * @param {any} svg
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param {any} allCommits
  */
-const drawArrow = (svg, commit1, commit2, allCommits) => {
-  const p1 = commitPos[commit1.id];
-  const p2 = commitPos[commit2.id];
-  const overlappingCommits = hasOverlappingCommits(commit1, commit2, allCommits);
-  // log.debug('drawArrow', p1, p2, overlappingCommits, commit1.id, commit2.id);
+const drawArrow = (svg, commitA, commitB, allCommits) => {
+  const p1 = commitPos[commitA.id]; // arrowStart
+  const p2 = commitPos[commitB.id]; // arrowEnd
+  const branchToGetCurve =
+    dir === 'TB'
+      ? p1.x < p2.x
+        ? commitB.branch
+        : commitA.branch
+      : p1.y < p2.y
+      ? commitB.branch
+      : commitA.branch;
```

Comment:
Should not this code be a part of `shouldRerouteArrow` function? It is not used anywhere else. Would not comparison of `seq` do effectively the same as comparison of x / y?

Reply:
Hi, thanks for taking a look. I originally wanted to keep the number of new arguments passed to `shouldRerouteArrow` to a minimum. But you're right: I could pass it `p1` and `p2` and do the calculation for `branchToGetCurve` inside `shouldRerouteArrow`. I can try that out.

I don't think comparing `seq` would be the same -- that would only tell you whether the one commit comes before or after the other on the branch, but here we need to know whether one commit is above or below the other (or in `TB` to the left or right), to know whether the normal branch curve will be a problem or cause a conflict that needs rerouting.

Reply:
I've made that change now although the coverage is decreased again, and I can't work out why. It seems to be arbitrary... :thinking:

Reply:
@nirname I just can't understand why coverage seems to drop randomly...

The number of files and the percentage change seems to be different each time and have no relation to my changes. Sometimes it is up, sometimes it is down.

Any ideas?

![Coverage decrease across indirectly affected files](https://github.com/mermaid-js/mermaid/assets/1826451/ee553adc-f05b-42a6-b241-adfbe2b783cc)

Reply:
No worries about coverage. We will have to investigate it further, but not withing this task. The main idea is that your code is covered by tests.

Reply:
@guypursey Why would not seq tell us the order of commits? I mean, whether the go left to right, is there any situation when we have 2 commits with the same position? My thoughts that x and y coordinates are simply the reflection of seq...

Reply:
I have not debugged it personally, so I can only imply that x and y depend on seq. Also we can get positions from within the function itself, so your initial idea of keeping minimum amount of arguments is still present.